crates/consensus/src/consensus/header_validation.rs

```diff
@@ -38,8 +38,13 @@ pub fn assert_header<'a>(
             FixedDecimal::from(5u64) / FixedDecimal::from(100u64);
         let c = (FixedDecimal::from(1u64) - active_slots_coeff).ln();
         let block_validator = BlockValidator::new(header, cbor, ledger, epoch_nonce, &c);
-        block_validator.validate().or_panic()?;
+        block_validator
+            .validate()
+            .map_err(|e| {
+                warn!("fail to validate header {}: {:?}", header.hash(), e);
```

Comment:
```suggestion
                warn!(header.hash = ?header.hash(), validation.error = ?e, "failed_to_validate");
```

A "best-practice" I've noted from using the `tracing` library (but which apply to telemetry as a whole) is to leverage fields as much as possible, and avoid embedding fields/payloads in the log message. So said differently, embrace _structured logs/traces_ ! This is what allows us to have nice JSON logs out-of-the-box 😛

Reply:
Totally agreed, that's why I voluntarily wrote this temporary kludge as a textual log entry so that we spot it more easily and change it to something consistent with the rest. 

In general, I think we should only ever need `instrument!`, any other kind of logs is probably unneeded, or is begging for a function to instrument :)

Reply:
> I think we should only ever need instrument!

That's a good rule of thumb regarding how we slice the code, indeed.